File: app/src/main/java/net/onamap/android/model/StateData.java
package net.onamap.android.model;

import java.io.Serializable;

public class StateData implements Serializable{
    public int icon;
    public int drawableResThumb;
    public String fullName;
    public String abbv;
    public boolean selected;

    public StateData(String abbv, String fullName, int drawableResThumb, int icon) {
        super();
        this.icon = icon;
        this.fullName = fullName;
        this.drawableResThumb = drawableResThumb;
        this.abbv = abbv;
        this.selected = false;
    }

    public void toggle() {
        selected = !selected;
    }
}

File: app/src/main/java/net/onamap/android/model/StateData.kt
package net.onamap.android.model

import java.io.Serializable

class StateData(
    val abbv: String,
    val fullName: String,
    val drawableResThumb: Int,
    val icon: Int
) : Serializable {
    var selected = false
    fun toggle() {
        selected = !selected
    }

}